drivers/storage/portworx/component/prometheus.go

```diff
@@ -656,9 +657,37 @@ func (c *prometheus) createPrometheusInstance(
 		prometheusInst.Spec.SecurityContext = &v1.PodSecurityContext{
 			RunAsNonRoot: &runAsNonRoot,
 			RunAsUser:    &runAsUser,
+			RunAsGroup:   &runAsUser,
+			FSGroup:      &runAsUser,
 		}
 	}
 
+	if cluster.Spec.Monitoring.Prometheus.Replicas != nil {
+		prometheusInst.Spec.Replicas = cluster.Spec.Monitoring.Prometheus.Replicas
+	} else {
+		prometheusInst.Spec.Replicas = &replicas
+	}
+
+	if cluster.Spec.Monitoring.Prometheus.Retention != "" {
+		prometheusInst.Spec.Retention = cluster.Spec.Monitoring.Prometheus.Retention
+	}
+
+	if cluster.Spec.Monitoring.Prometheus.RetentionSize != "" {
+		prometheusInst.Spec.RetentionSize = cluster.Spec.Monitoring.Prometheus.RetentionSize
+	}
+
+	if cluster.Spec.Monitoring.Prometheus.Storage != nil {
```

Comment:
we should do some sort of validation here, e.g. if storage is specified but not volume/mount, should we raise an error? or let the prometheus operator to handle everything?

Reply:
If the client doesn't specify ```storage```, prometheus is going to use default value (i.e. no storage/volumeMounts). Later we can add fallback storage if explicitly required.

Reply:
I mean when `storage` is specified, is volume and mount required as well? assume if user has storage and volume specified but not mount, what would happen in that case?

Reply:
Volume and Volumemount are not required. 
If we use automatic provisioning of storage by specifying the storageclass in ```storage```(StorageSpec) section. When creating the Prometheus object, a PersistentVolumeClaim is used for each Pod in the StatefulSet, and the storage should automatically be provisioned, mounted and used.
Volumes allows configuration of additional volumes on the output StatefulSet definition. Volumes specified will be appended to other volumes that are generated as a result of StorageSpec objects.
In my test, volume ```prometheus-px-prometheus-db``` is provisioned with storageclass ```px-repl3-sc```. Volume ```testvlm``` is additional volume configured through ```Volumes```.